fedora_messaging/_session.py

```diff
@@ -167,10 +167,13 @@ def publish(self, message, exchange=None):
                 _log.error(str(e))
                 if self._connection and self._connection.is_open:
                     self._connection.close()
+                self._connection = self._channel = None
                 raise ConnectionException(reason=e)
         except pika_errs.AMQPError as e:
+            _log.info("Resetting connection to %s", self._parameters.host)
             if self._connection and self._connection.is_open:
                 self._connection.close()
+            self._connection = self._channel = None
```

Comment:
The code here looks similar to the code above - would it be helpful to have a _reset_connection() to consolidate?

Reply:
Yeah, but I'd rather get rid of this entire module in the medium term and run everything in Twisted with https://crochet.readthedocs.io/en/stable/. Dealing with AMQP in an asynchronous fashion is much simpler.